torch/__init__.py

```diff
@@ -384,6 +392,20 @@ def __sym_not__(self) -> "SymBool":
     def __repr__(self):
         return str(self.node)
 
+    def __eq__(self, other):
+        if self.node.is_constant():
+            if isinstance(other, builtins.bool):
+                return self.node.bool_() == other
+            elif isinstance(other, SymBool) and other.node.is_constant():
+                return self.node.bool_() == other.node.bool_()
+        return self is other
+
+    def __hash__(self):
+        if self.node.is_constant():
+            return hash(self.node.bool_())
+        else:
+            return object.__hash__(self)
```

Comment:
I'd prefer if we throw an error here, as we did previously

Reply:
Technically we didn't throw an error here previously!
The default behavior for python classes that don't implement __eq__ or __hash__  is to some kind of object identity equality and hashing.
And so, SymBool doesn't error because it does not implement __eq__(why?), but SymFloat and SymInt do error, because they eq IS implemented.
That said, erroring here might be better for symmetry, so that sounds OK to me.

Reply:
That's a bug 🤡

Reply:
Yes if you could fix this up while you are at it that would be great